### PR TITLE
Contribution guidelines

### LINKED ISSUE #105: Add contribution guidelines
Add contribution guidelines as a separate file named `contribution_guidelines.md` and reference `contribution_guidelines.md` in `README.md`.

### PR DESCRIPTION
**Pull request by JMorado**
Closes #105 
- [x] Add contribution guidelines as a separate file named contribution_guidelines.md.
- [x] Reference contribution_guidelines.md in README.md.

### CODE REVIEW

File: contribution_guidelines.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+1. Add changes to the staging area with `git add src/pybdy/file_change.py`
+1. Commit changes: `git commit -m "#103 Remove pynemo traces"`
+1. As hooks are run on every commit because before we executed `pre-commit install`, they may flag problems with the code and/or change some of its parts, thus preventing the commit attempt to be successful. If all pre-commit hooks passed, jump to step 7.
+1. Solve any issue flagged by the pre-commit hooks.
+1. Update already stag files executing `git add -u`.
+1. Commit changes once again `git commit -m "#103 Remove pynemo traces"`.
+1. Finally, push changes to GitHub and set the remote as upstream with `git push --set-upstream origin issue-103-remove-pynemo-traces`.
+
+## Make a pull request
+
```
Comment (jpolton) [inline contribution_guidelines.md:46]: Should there be a "merge master into branch" step here, or maybe at the start of the Pre-push steps?

Comment (JMorado) [reply contribution_guidelines.md:46]: I think it is better to merge a branch into master (and solve any conflicts that may arise at that point) than the master into a branch. In this way, we are sure that the upstream changes don't break a PR. 

Maybe something along the lines of

```
git checkout master
git pull
```
before creating a branch could help to ensure that whatever branch is created, it starts from the current HEAD of the upstream master.

Comment (jpolton) [reply contribution_guidelines.md:46]: My concern would be more about a PR breaking the master, than the other way around.
(With the workflow ordering you propose, I do see how the qa and unit testing would be applied as the conflicts would be resolved on the GitHub server?)

BTW I'm happy to try your proposal and learn by doing. I consider this issue "resolved"

Comment (JMorado) [reply contribution_guidelines.md:46]: OK, I got your point now.

So yes, if we want the developers to be responsible for resolving any potential conflicts, then they should merge master into branch before pushing. If the PR has no conflicts, I think there is no point in merging as it just introduces a redundant commit. 

I will adapt the guidelines to introduce these ideas.

**Review by jpolton — APPROVED** [review]

**Pull request merged**